frontend/src/components/Pages/Todo/TodoList.jsx

```diff
@@ -1,23 +1,32 @@
 import "./todo.css";
 import React from "react";
 import TodoItem from "./TodoItem";
+import TodoItemGroup from "./TodoItemGroup";
 
-export function renderItems(items, ifEmpty = undefined) {
-  var index = 0;
-  if (items.length !== 0) {
-    return items.map((todoItem) => (
-      <TodoItem key={index++} itemData={todoItem} />
-    ));
-  }
-  return ifEmpty && <li>{ifEmpty}</li>;
+export function renderItems(items, { ifEmpty, onItemToggled }) {
+  return (
+    <ul>
+      {items.length ? (
+        items.map((todoItem, index) => {
+          const hasNestedItems = todoItem.items && todoItem.items.length;
+          if (hasNestedItems)
```

Comment:
`todoItem.items && todoItem.items.length` could be simplified down to just `todoItem.items`

Reply:
What if you had an object like:
```javascript
const todoItem = {
    items: []
};
```
Wouldn't `todoItem.items` return true here, as `todoItem.items` is defined, despite it being empty?

Reply:
Ah mb, I meant to simplify it to `todoItem.items.length`

Reply:
Hmm, I'm not sure I can guarantee `items` will be defined though, but I suppose it could probably be simplified to `todoItem.items?.length`.

Reply:
Good idea, that will work

Reply:
Sweet, I'll quickly make that change :)